From 52aa316543bdd56e945997bb7b73d093c8134b57 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:07:08 -0400
Subject: [powerpc] kernel: Use %12.12s instead of %12s to avoid memory overflow

Message-id: <1410545655-205645-200-git-send-email-dzickus@redhat.com>
Patchwork-id: 94489
O-Subject: [RHEL7 PATCH 199/626] arch/powerpc/kernel: Use %12.12s instead of %12s to avoid memory overflow
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit e0513d9ea8dd2a7b21b699ada8d59d8afbb8b5b7
Author: Chen Gang <gang.chen@asianux.com>
Date:   Thu Jan 24 12:14:52 2013 +0800

    arch/powerpc/kernel: Use %12.12s instead of %12s to avoid memory overflow

    for tmp_part->header.name:
        it is "Terminating null required only for names < 12 chars".
        so need to limit the %.12s for it in printk

      additional info:

        %12s  limit the width, not for the original string output length
              if name length is more than 12, it still can be fully displayed.
              if name length is less than 12, the ' ' will be filled before name.

        %.12s truly limit the original string output length (precision)

    Signed-off-by: Chen Gang <gang.chen@asianux.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/nvram_64.c b/arch/powerpc/kernel/nvram_64.c
index fd82c28..28b898e 100644
--- a/arch/powerpc/kernel/nvram_64.c
+++ b/arch/powerpc/kernel/nvram_64.c
@@ -210,7 +210,7 @@ static void __init nvram_print_partitions(char * label)
  printk(KERN_WARNING "--------%s---------\n", label);
  printk(KERN_WARNING "indx\t\tsig\tchks\tlen\tname\n");
  list_for_each_entry(tmp_part, &nvram_partitions, partition) {
-  printk(KERN_WARNING "%4d    \t%02x\t%02x\t%d\t%12s\n",
+  printk(KERN_WARNING "%4d    \t%02x\t%02x\t%d\t%12.12s\n",
          tmp_part->index, tmp_part->header.signature,
          tmp_part->header.checksum, tmp_part->header.length,
          tmp_part->header.name);
-- 
1.7.1